wgsl/index.bs

```diff
@@ -132,13 +132,13 @@ TODO: What indicates the end of a line?  (E.g. A line ends at the next linefeed
   <thead>
     <tr><th>Token<th>Definition
   </thead>
-  <tr><td>`FLOAT_LITERAL`<td>`(-?[0-9]*.[0-9]+ | -?[0-9]+.[0-9]*)((e|E)(+|-)?[0-9]+)?`
+  <tr><td>`FLOAT_LITERAL`<td>`-?(([0-9]*.[0-9]+ | [0-9]+.[0-9]*((e|E)(+|-)?[0-9]+)?) | 0x([0-9a-fA-F]*.?[0-9a-fA-F]+ | [0-9a-fA-F]+.[0-9a-fA-F]*)(p|P)(+|-)?[0-9]+)`
```

Comment:
This is really dense.
I suggest breaking this up into two,  DECIMAL_FLOAT_LITERAL  and HEX_FLOAT_LITERAL.

Then in the 3 other places where  FLOAT_LITERAL is used, replace it by uses of both of those.

Reply:
This section needs more exposition about how to actually convert these literals into actual values.
I could take that as a followup.